r/R/cell_cycle_scoring.R

```diff
@@ -0,0 +1,132 @@
+#' CellCyCleScoring
+#'
+#' @param req
+#' @param data
+#'
+#' @return Formatted cellsets
+#' @export
+#'
+#' @examples
+cellCycleScoring <- function(req, scdata) {
+  cellSets <- run_cell_cycle_scoring(scdata)
+
+  message("formatting cellsets")
+
+  formatted_cell_sets <-
+    format_cluster_cellsets(cellSets, "Phase", scdata@misc$color_pool)
+  message("updating through api")
+
+  updateCellSetsThroughApi(
+    formatted_cell_sets,
+    req$body$apiUrl,
+    req$body$experimentId,
+    "Phase",
+    req$body$authJwt,
+    append = FALSE
+  )
+
+  return(formatted_cell_sets)
+}
+
+#' run_cell_cycle_scoring
+#'
+#' @param scdata
+#' @param req
+#'
+#' @return
+#' @export
+#'
+#' @examples
+run_cell_cycle_scoring <- function(scdata, req) {
+  message("Running Cell Cycle Scoring")
+
+  s.gene.names <- Seurat::cc.genes$s.genes
+  g2m.gene.names <- Seurat::cc.genes$g2m.genes
+
+  s.gene.ids <-
+    na.omit(scdata@misc$gene_annotations[match(tolower(s.gene.names), tolower(scdata@misc$gene_annotations$name)), "input"])
+  g2m.gene.ids <-
+    na.omit(scdata@misc$gene_annotations[match(tolower(g2m.gene.names), tolower(scdata@misc$gene_annotations$name)), "input"])
+
+  s.genes <- c(s.gene.names,s.gene.ids)
+  g2m.genes <- c(g2m.gene.names,g2m.gene.ids)
+
+  phase <- tryCatch({
+    Seurat::CellCycleScoring(
+      scdata,
+      s.features = s.genes,
+      g2m.features = g2m.genes,
+      set.ident = TRUE
+    )@meta.data$Phase
+  }, error = function(err) {
+    rep("Undetermined", ncol(scdata))
```

Comment:
would this error message be clear enough?

Reply:
It's not an error message though, it will just create a single "undetermined" cellset.

Reply:
Maybe we should use a different word here. What do you think?

Reply:
maybe something more explicit? cc_phase_undetermined? or something like that. We will not be throwing an error, and a new cellset will be created. That cellset should have a clear name, so that when you see it in data exploration you know where it comes from.